### PR TITLE
Replaced usages of schema_seq_no to triple(name, version,did)

### PR DESCRIPTION
**Pull request by Artemkaaas**
Signed-off-by: artem.ivanov <artem.ivanov@dsr-company.com>

### CODE REVIEW

File: libindy/src/api/anoncreds.rs [hunk 1]
```diff
@@ -65,19 +63,18 @@ pub extern fn indy_issuer_create_and_store_claim_def(command_handle: i32,
...
 /// Create a new revocation registry for the given claim definition.
-/// Stores it in a secure wallet identifying by the returned key.
+/// Stores it in a secure wallet.
 ///
 /// #Params
 /// wallet_handle: wallet handler (created by open_wallet).
 /// command_handle: command handle to map callback to user context.
 /// issuer_did: a DID of the issuer signing revoc_reg transaction to the Ledger
-/// schema_seq_no: seq no of a schema transaction in Ledger
+/// schema_json: schema as a json
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Why didn't we need it before?

File: libindy/src/api/anoncreds.rs [hunk 2]
```diff
@@ -87,12 +84,13 @@ pub extern fn indy_issuer_create_and_store_claim_def(command_handle: i32,
 pub extern fn indy_issuer_create_and_store_revoc_reg(command_handle: i32,
                                                      wallet_handle: i32,
                                                      issuer_did: *const c_char,
-                                                     schema_seq_no: i32,
+                                                     schema_json: *const c_char,
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Why didn't we need it before?

File: libindy/src/api/anoncreds.rs [hunk 3]
```diff
@@ -101,7 +99,7 @@ pub extern fn indy_issuer_create_and_store_revoc_reg(command_handle: i32,
                 IssuerCommand::CreateAndStoreRevocationRegistry(
                     wallet_handle,
                     issuer_did,
-                    schema_seq_no,
+                    schema_json,
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Why didn't we need it before?

File: libindy/src/api/anoncreds.rs [hunk 4]
```diff
@@ -113,23 +111,24 @@ pub extern fn indy_issuer_create_and_store_revoc_reg(command_handle: i32,
...
 /// an stored into the wallet.
 ///
 /// #Params
 /// wallet_handle: wallet handler (created by open_wallet).
 /// command_handle: command handle to map callback to user context.
 /// claim_req_json: a claim request with a blinded secret
-///     from the user (returned by prover_create_and_store_claim_req).
-///     Also contains schema_seq_no and issuer_did
+/// from the user (returned by prover_create_and_store_claim_req).
+/// Also contains schema_seq_no and issuer_did
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: it contains not schema_seq_no, but schema_key (triple)

File: libindy/src/api/anoncreds.rs [hunk 5]
```diff
@@ -113,23 +111,24 @@ pub extern fn indy_issuer_create_and_store_revoc_reg(command_handle: i32,
...
 ///     Example:
 ///     {
 ///      "blinded_ms" : <blinded_master_secret>,
-///      "schema_seq_no" : <schema_seq_no>,
-///      "issuer_did" : <issuer_did>
+///      "schema_key" : {name: string, version: string, did: string},
+///      "issuer_did" : string
+///      "prover_did" : string
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Why do we need prover DID here?

File: libindy/src/api/anoncreds.rs [hunk 6]
```diff
@@ -386,8 +387,9 @@ pub extern fn indy_prover_create_master_secret(command_handle: i32,
 /// Claim request json.
 ///     {
 ///      "blinded_ms" : <blinded_master_secret>,
-///      "schema_seq_no" : <schema_seq_no>,
-///      "issuer_did" : <issuer_did>
+///      "schema_key" : {name: string, version: string, did: string},
+///      "issuer_did" : string,
+///      "prover_did" : string
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Why do we need prover_did?

File: libindy/src/api/anoncreds.rs [hunk 7]
```diff
@@ -706,12 +712,15 @@ pub extern fn indy_prover_get_claims_for_proof_req(command_handle: i32,
...
-///         "aggregated_proof": <aggregated_proof>
+///         "proof": {
+///             "proofs": {
+///                 "claim_proof1_referent": <claim_proof>,
+///                 "claim_proof2_referent": <claim_proof>,
+///                 "claim_proof3_referent": <claim_proof>
+///             },
+///             "aggregated_proof": <aggregated_proof>
+///         }
+///         "identifiers": [{issuer_did, rev_reg_seq_no, schema_key: {name, version, did}}]
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Each part of the proof may belong to different schemas and issuers. Why did we move this information out of `"claim_proofX_referent"`?

File: libindy/src/api/anoncreds.rs (hunk 6 above)
Comment (Artemkaaas) [reply libindy/src/api/anoncreds.rs]: prover_did is used in Issuer::sign_claim function

File: libindy/src/api/anoncreds.rs (hunk 1 above)
Comment (Artemkaaas) [reply libindy/src/api/anoncreds.rs]: We used to use issuer_did:schema_seq_no to identify rev_reg inside wallet. Now we have following key issuer_did:(schema_key)

File: libindy/src/api/anoncreds.rs (hunk 7 above)
Comment (Artemkaaas) [reply libindy/src/api/anoncreds.rs]: Changed identifiers format to
"identifiers": {"claim_proof1_referent":{issuer_did, rev_reg_seq_no, schema_key: {name, version, did}}}

File: libindy/src/api/anoncreds.rs [hunk 8]
```diff
@@ -685,7 +691,7 @@ pub extern fn indy_prover_get_claims_for_proof_req(command_handle: i32,
 ///         "value": requested value of attribute
 ///         "restrictions": [
 ///             {
-///                 "schema_seq_no": int, (Optional)
+///                 "schema_key": {name (Optional), version (Optional), did (Optional)}, (Optional)
```
Comment (ashcherbakov) [inline libindy/src/api/anoncreds.rs]: Please add a comment about pattern matching for requested attributes in proof request (that it's case insensitive and ignore spaces). Add the comment in all methods dealing with proof request and pattern matching

File: libindy/tests/anoncreds.rs [hunk 9]
```diff
@@ -128,39 +110,52 @@ mod high_cases {
...
+            assert_eq!(claim_offers.len(), 2);
+            assert!(claim_offers.contains(&ClaimOffer { issuer_did: ISSUER_DID.to_string(), schema_key: AnoncredsUtils::gvt_schema_key() }));
+            assert!(claim_offers.contains(&ClaimOffer { issuer_did: DID.to_string(), schema_key: AnoncredsUtils::gvt_schema_key() }));
+        }
+
+        #[test]
+        fn prover_get_claim_offers_works_for_filter_by_part_of_schema() {
+            let (wallet_handle, _) = AnoncredsUtils::init_common_wallet();
+
+            let claim_offers = AnoncredsUtils::prover_get_claim_offers(wallet_handle, r#"{"schema_key":{"name":"gvt"}}"#).unwrap();
```
Comment (ashcherbakov) [inline libindy/tests/anoncreds.rs]: Please make sure that we can use any part of the Schema (name or verion or did)

File: libindy/tests/anoncreds.rs [hunk 10]
```diff
@@ -384,31 +370,44 @@ mod high_cases {
...
+            assert_eq!(claims[0].schema_key, AnoncredsUtils::gvt_schema_key());
+            assert_eq!(claims[1].schema_key, AnoncredsUtils::gvt_schema_key());
+        }
+
+
+        #[test]
+        fn prover_get_claims_works_for_filter_by_part_of_schema() {
+            let (wallet_handle, _) = AnoncredsUtils::init_common_wallet();
+
+            let claims = AnoncredsUtils::prover_get_claims(wallet_handle, r#"{"schema_key":{"name":"gvt"}}"#).unwrap();
```
Comment (ashcherbakov) [inline libindy/tests/anoncreds.rs]: Please add tests to make sure that we can use any part of the Schema (name or verion or did)

**Pull request merged**